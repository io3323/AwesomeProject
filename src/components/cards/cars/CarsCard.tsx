import React from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import {styles} from './styles';
import {ContextTitle} from '../../../UIKit/context/ContextTitle';

export const CarsCard = ({
  id,
  model,
  year,
  engine,
  transmission,
  price,
  color,
  topSpeed,
  sectionNum,
  setSelectedId,
  setMoreInfoModalVisible,
}) => {
  return (
    <TouchableOpacity
      onPress={() => {
        setSelectedId(id);
        setMoreInfoModalVisible(true);
      }}
      style={[
        styles.main,
        {
          width: sectionNum === 1 ? '100%' : sectionNum === 2 ? '50%' : '25%',
          height: sectionNum === 1 ? 150 : sectionNum === 2 ? 120 : 120,
          marginVertical: sectionNum === 1 ? 40 : sectionNum === 2 ? 10 : 10,
          paddingHorizontal: sectionNum === 1 ? 20 : sectionNum === 2 ? 10 : 10,
        },
      ]}>
      <View style={styles.container}>
        <ContextTitle title={'Модель'} description={model} />
        <ContextTitle title={'Год выпуска'} description={year} />
        {sectionNum === 1 && (
          <ContextTitle title={'Двигатель'} description={engine} />
        )}
        {sectionNum === 1 && (
          <ContextTitle title={'Тип трансмисии'} description={transmission} />
        )}
        {sectionNum === 1 && (
          <ContextTitle
            title={'Максимальная скорость'}
            description={topSpeed}
          />
        )}
        <ContextTitle title={'Цена'} description={price} />
        <ContextTitle title={'Цвет'} description={color} />
      </View>
    </TouchableOpacity>
  );
};
