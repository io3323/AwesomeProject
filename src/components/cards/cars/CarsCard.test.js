import React from 'react';
import renderer from 'react-test-renderer';
import {CarsCard} from './CarsCard';

test('CarsCard Component', () => {
  // Mock props
  const props = {
    id: 1,
    model: 'Test Model',
    year: 2023,
    engine: '2.0L Turbocharged',
    transmission: 'Automatic',
    price: '$25,000',
    color: 'Red',
    topSpeed: '200 km/h',
    sectionNum: 1,
    setSelectedId: jest.fn(),
    setMoreInfoModalVisible: jest.fn(),
  };

  // Render the component
  const component = renderer.create(<CarsCard {...props} />);

  // Get the JSON representation of the component
  const tree = component.toJSON();

  // Expect the component to match the snapshot
  expect(tree).toMatchSnapshot();

  // Simulate press
  tree.props.onPress();

  // Ensure that setSelectedId and setMoreInfoModalVisible were called with the correct parameters
  expect(props.setSelectedId).toHaveBeenCalledWith(props.id);
  expect(props.setMoreInfoModalVisible).toHaveBeenCalledWith(true);
});
