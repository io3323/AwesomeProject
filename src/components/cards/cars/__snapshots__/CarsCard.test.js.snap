// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarsCard Component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "height": 150,
      "marginVertical": 40,
      "opacity": 1,
      "paddingHorizontal": 20,
      "width": "100%",
    }
  }
>
  <View
    style={
      {
        "borderColor": "black",
        "borderRadius": 5,
        "borderStyle": "solid",
        "borderWidth": 1,
        "padding": 10,
      }
    }
  >
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text>
        Модель
      </Text>
      <Text>
        Test Model
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text>
        Год выпуска
      </Text>
      <Text>
        2023
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text>
        Двигатель
      </Text>
      <Text>
        2.0L Turbocharged
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text>
        Тип трансмисии
      </Text>
      <Text>
        Automatic
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text>
        Максимальная скорость
      </Text>
      <Text>
        200 km/h
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text>
        Цена
      </Text>
      <Text>
        $25,000
      </Text>
    </View>
    <View
      style={
        {
          "display": "flex",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginTop": 10,
        }
      }
    >
      <Text>
        Цвет
      </Text>
      <Text>
        Red
      </Text>
    </View>
  </View>
</View>
`;
